composer.json

```diff
@@ -21,24 +21,34 @@
     ],
     "require": {
         "php": "^8.0",
+        "ext-dom": "*",
         "ext-exif": "*",
         "ext-fileinfo": "*",
+        "ext-filter": "*",
         "ext-gd": "*",
+        "ext-hash": "*",
+        "ext-intl": "*",
+        "ext-json": "*",
         "api-platform/core": "^2.6",
         "babdev/pagerfanta-bundle": "^2.5",
         "behat/transliterator": "^1.3",
         "doctrine/collections": "^1.6",
         "doctrine/dbal": "^2.7",
         "doctrine/doctrine-bundle": "^1.12 || ^2.0",
         "doctrine/doctrine-migrations-bundle": "^3.0.1",
+        "doctrine/event-manager": "^1.1",
         "doctrine/inflector": "^1.4 || ^2.0",
         "doctrine/migrations": "^3.0",
         "doctrine/orm": "^2.7",
+        "doctrine/persistence": "^2.3",
         "egulias/email-validator": "^3.1",
         "fakerphp/faker": "^1.9",
         "friendsofphp/proxy-manager-lts": "^1.0",
         "friendsofsymfony/rest-bundle": "^3.0",
         "gedmo/doctrine-extensions": "^3.2",
+        "guzzlehttp/guzzle": "^6.5",
+        "guzzlehttp/psr7": "^1.8",
+        "hwi/oauth-bundle": "^1.4",
```

Comment:
This is a tough one. It's meant to be an optional dependency, but the moment I wrote these 2 words I remembered [this post](https://matthiasnoback.nl/2014/04/theres-no-such-thing-as-an-optional-dependency/) from Matthias Noback about optional dependencies, which I still agree with.
Keeping it in `require` might result in dependency issue on update for those who don't use the bundle.
I suggest to move it back to `require-dev`, ignore it in composer-require-checked config  and address the issue separately.
Perhaps this integration should move from Sylius Core to a plugin.

Reply:
We already talked about this topic in this same PR [here](https://github.com/Sylius/Sylius/pull/10664#discussion_r331455461).

I'm still for having `hwi/oauth-bundle` as a strict/hard dependency. Remember that not having it in the `require` section already caused issues to others (see [here](https://github.com/Sylius/Sylius/issues/10530#issuecomment-515063529)).

Moreover in the meantime `hwi/oauth-bundle` has become stable.

Reply:
I missed the first conversation, but I remember the second one.
I still think that `hwi/oauth-bundle` should not be in the `require` section of `sylius/sylius` and that we need an alternative solution for it, like extracting this feature in a plugin.